components/datetime/src/provider/date_time.rs

```diff
@@ -153,10 +151,13 @@ where
             data_provider,
             date_patterns_data,
             locale,
+            cal_val: None,
         };
         match options {
             DateTimeFormatterOptions::Length(bag) => selector
                 .pattern_for_length_bag(bag, Some(preferences::Bag::from_data_locale(locale))),
+            #[allow(clippy::panic)] // explicit panic in experimental mode
+            _ => panic!("Non-experimental constructor cannot handle experimental options"),
```

Comment:
Why are the tests hitting this?

Reply:
Because they are using the non-experimental constructors with an (experimental) components bag option

Reply:
But the tests don't use any `_experimental` functions. The point of this split is to isolate experimental code so that even if the experimental feature is accidentally turned on, the behaviour of the non-experimental API doesn't change. That doesn't seem to be the case.

Reply:
The reason this panic exists is because I'm reusing the same `DateTimeFormatterOptions` enum and adding an additional variant to the enum in experimental mode, which non-experimental mode can't handle.

The only alternative would be to make an entirely standalone `DateTimeFormatterExperimentalOptions`, which is doable but still has a lot of integration challenge because the tests are written to assume a single options type.

I would rather land this as-is because I feel it's an improvement over the status quo. But if you think this detracts from the status quo, I'll close the PR.